app/controllers/DeleteDisclosureSummaryController.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.CrossBorderArrangementsConnector
+import controllers.actions._
+import javax.inject.Inject
+import pages.{GeneratedIDPage, URLPage, ValidXMLPage}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import renderer.Renderer
+import repositories.SessionRepository
+import services.XMLValidationService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import utils.CheckYourAnswersHelper
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.xml.Elem
+
+class DeleteDisclosureSummaryController @Inject()(
+    override val messagesApi: MessagesApi,
+    identify: IdentifierAction,
+    getData: DataRetrievalAction,
+    requireData: DataRequiredAction,
+    sessionRepository: SessionRepository,
+    xmlValidationService: XMLValidationService,
+    crossBorderArrangementsConnector: CrossBorderArrangementsConnector,
+    val controllerComponents: MessagesControllerComponents,
+    renderer: Renderer
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      request.userAnswers.get(URLPage) match {
+        case Some(url) =>
+          val xml: Elem = xmlValidationService.loadXML(url)
+          val helper = new CheckYourAnswersHelper(request.userAnswers)
+          val fileToDelete = helper.displaySummaryFromInstruction(xml)
```

Comment:
Ok, so now you've got the Dac6MetaData you have everything you need from the xml. Refactor the displaySummaryFromInstruction to take a dac6MetaData class and get everything from that.

Reply:
you'll have to retrieve that from the useranswers ofcourse

Reply:
Yep - Forgot this was part of reason we were making this change. Changed it now - hoping this is ok.